packages/lesswrong/server/repos/TagsRepo.ts

```diff
@@ -50,4 +50,31 @@ export default class TagsRepo extends AbstractRepo<DbTag> {
     const {count} = await this.getRawDb().one(`SELECT COUNT(*) FROM "Tags"`);
     return count;
   }
+
+  async getUserTopTags(userId: string, limit = 15): Promise<UserTopTag[]> {
+    return this.getRawDb().any(`
+      SELECT
+        public."Tags".name,
+        public."TagRels"."tagId",
+        COUNT(*) AS count
+      FROM public."TagRels"
+      INNER JOIN public."Tags" ON public."TagRels"."tagId" = public."Tags"._id
+      WHERE
+        public."TagRels"."postId" IN (
+          SELECT public."Comments"."postId"
+          FROM public."Comments"
+          WHERE public."Comments"."userId" = $1
+          AND public."Comments"."postedAt" > NOW() - INTERVAL '3 years'
+        )
+        AND public."Tags".name NOT IN (
+          'Community', 'Rationality', 'World Modeling', 'Site Meta', 'Covid-19', 'Practical', 
+          'World Optimization', 'Best of LessWrong', 'LessWrong Review', 'LessWrong Event Transcripts', 
+          'Existential Risk', 'AI Risk', 'Epistemic Review', 'Open Threads', 'AI', 'Politics', 'Epistemology',
+          'Drama', 'Meta', 'Has Diagram', 'News', 'LW Moderation', 'Experiments', 'Dialogue (format)', 'LW Team Announcements'
+        )
+      GROUP BY public."TagRels"."tagId", public."Tags".name
```

Comment:
You don't need the schema qualification (`public.`), but should alias the table names